src/corehost/cli/hostpolicy/hostpolicy.cpp

```diff
@@ -418,26 +499,120 @@ namespace
             return StatusCode::LibHostInvalidArgs;
         }
     }
+
+    int get_property(
+        const context_handle instance,
+        const pal::char_t *key,
+        const pal::char_t **value)
+    {
+        if (instance == nullptr || key == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        hostpolicy_context_t *context = static_cast<hostpolicy_context_t*>(instance);
+        if (!context->coreclr_properties.try_get(key, value))
+            return StatusCode::HostPropertyNotFound;
+
+        return StatusCode::Success;
+    }
+
+    int set_property(
+        const context_handle instance,
+        const pal::char_t *key,
+        const pal::char_t *value)
+    {
+        if (instance == nullptr || key == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        hostpolicy_context_t *context = static_cast<hostpolicy_context_t*>(instance);
+        if (value != nullptr)
+        {
+            context->coreclr_properties.add(key, value);
+        }
+        else
+        {
+            context->coreclr_properties.remove(key);
+        }
+
+        return StatusCode::Success;
+    }
+
+    int get_properties(
+        const context_handle instance,
+        size_t * count,
+        const pal::char_t **keys,
+        const pal::char_t **values)
+    {
+        if (instance == nullptr || count == nullptr)
+            return StatusCode::InvalidArgFailure;
+
+        hostpolicy_context_t *context = static_cast<hostpolicy_context_t*>(instance);
+        size_t actualCount = context->coreclr_properties.count();
+        if (*count < actualCount || keys == nullptr || values == nullptr)
+        {
+            *count = actualCount;
```

Comment:
We need to set the count always - if the input count is larger than actual count there would be no way for the caller to tell how many properties were filled in.